src/commands/register.ts

```diff
@@ -2,142 +2,161 @@ import { flags } from '@oclif/command';
 import chalk from 'chalk';
 import { classToPlain } from 'class-transformer';
 import { cli } from 'cli-ux';
-import inquirer from 'inquirer';
+import fs from 'fs-extra';
 import path from 'path';
 import untildify from 'untildify';
 import Command from '../base-command';
 import LocalDependencyManager from '../common/dependency-manager/local-manager';
 import MissingContextError from '../common/errors/missing-build-context';
 import { buildImage, getDigest, pushImage, strip_tag_from_image } from '../common/utils/docker';
-import { ServiceNode } from '../dependency-manager/src';
+import { ServiceConfig, ServiceNode } from '../dependency-manager/src';
+import { ComponentConfigBuilder } from '../dependency-manager/src/component-config/builder';
 
-export interface CreateServiceVersionInput {
+export interface PutComponentVersionDto {
   tag: string;
-  digest: string;
   config: any;
 }
 
-export default class ServiceRegister extends Command {
-  static aliases = ['service:register', 'services:register', 'svcs:register', 'svc:register'];
-  static description = 'Register a new Service with Architect Cloud';
+export default class ComponentRegister extends Command {
+  static aliases = ['component:register', 'components:register', 'c:register', 'comp:register'];
+  static description = 'Register a new Component with Architect Cloud';
 
   static flags = {
     ...Command.flags,
-    services: flags.string({
```

Comment:
Should this still exist if someone wants to build/push only one service of the component rather than all of them? That could probably save some people a lot of time in CI jobs and when iterating on individual services

Reply:
This is an interesting comment. The backend as we currently designed it I don't think would support this... _unless_ we wrapped the `Service` in a component and persisted it _as a_ `Component`.

I'd be curious to get @tjhiggins 's thoughts.

But the short answer is: I think this flag would be nonsensical for now without some reconsideration of the API.

Reply:
I think the best approach for us long term is to treat our components as OCI compliant images. OCI registries already solve build-time optimization and partial builds via layering meaning we wouldn't need the user to do anything in particular for this. We'd just need components to be conformant (which they aren't right now) so we can store them in a registry. I'd recommend we avoid any obligations on the users side with the working assumption that we will do something like that one day.